QuickOSM/core/process.py

```diff
@@ -165,13 +170,23 @@ def open_file(
             if config_outputs and config_outputs[layer]['style']:
                 new_layer.loadNamedStyle(config_outputs[layer]['style'])
             else:
-                # Loading default styles
-                if layer == "multilinestrings" or layer == "lines":
-                    if "colour" in item['tags']:
-                        new_layer.loadNamedStyle(
-                            join(dirname(dirname(abspath(__file__))),
-                                 "styles",
-                                 layer + "_colour.qml"))
+                if "colour" in item['tags']:
+                    index = item['tags'].index('colour')
+                    colors = new_layer.uniqueValues(index)
+                    categories = []
+                    for value in colors:
+                        if layer in ['lines', 'multilinestrings']:
+                            symbol = QgsSymbol.defaultSymbol(QgsWkbTypes.LineGeometry)
+                        elif layer == "points":
+                            symbol = QgsSymbol.defaultSymbol(QgsWkbTypes.PointGeometry)
+                        elif layer == "multipolygons":
+                            symbol = QgsSymbol.defaultSymbol(QgsWkbTypes.MultiPolygon)
+                        symbol.setColor(QColor(value))
```

Comment:
What happen if the color is not valid ?
Maybe add a rule to black if not QColor.isValid() ?

Also, maybe all OSM features doesn't have a color, remember, a feature in OSM might not have the color defined.
So do you have a fallback color ? (like ELSE" in rule based)

Reply:
There is always an color for 'other'